control-plane/pkg/reconciler/channel/channel.go

```diff
@@ -129,10 +132,9 @@ func (r *Reconciler) reconcileKind(ctx context.Context, channel *messagingv1beta
 	}
 
 	// get security option for Sarama with secret info in it
-	saramaSecurityOption, err := r.saramaSecurityOption()
-	if err != nil {
-		return fmt.Errorf("failed to create security (auth) option: %w", err)
-	}
+	saramaSecurityOption := security.NewOptionFromSecret(secret)
```

Comment:
nit: before it was `saramaSecurityOption` 

migth it be hand to give a bit of context to the very generic name of `NewOptionFromSecret`  ?

Reply:
Ok, let's do `NewSaramaSecurityOptionFromSecret`?